packages/builders/src/messages/embed/UnsafeEmbed.ts

```diff
@@ -4,110 +4,163 @@ import type {
+	public get color() {
+		return this.data.color;
+	}
 
 	/**
 	 * The timestamp of the embed in the ISO format
 	 */
-	public readonly timestamp?: string;
+	public get timestamp() {
+		return this.data.timestamp;
+	}
 
 	/**
 	 * The embed thumbnail data
 	 */
-	public readonly thumbnail?: APIEmbedThumbnail;
+	public get thumbnail(): EmbedImageData | undefined {
+		if (!this.data.thumbnail) return undefined;
+		return {
+			url: this.data.thumbnail.url,
+			proxyURL: this.data.thumbnail.proxy_url,
+			height: this.data.thumbnail.height,
+			width: this.data.thumbnail.width,
+		};
+	}
 
 	/**
 	 * The embed image data
 	 */
-	public readonly image?: APIEmbedImage;
+	public get image(): EmbedImageData | undefined {
+		if (!this.data.image) return undefined;
+		return {
+			url: this.data.image.url,
+			proxyURL: this.data.image.proxy_url,
+			height: this.data.image.height,
+			width: this.data.image.width,
+		};
+	}
 
 	/**
 	 * Received video data
 	 */
-	public readonly video?: APIEmbedVideo;
+	public get video(): APIEmbedVideo | undefined {
+		return this.data.video;
+	}
 
 	/**
 	 * The embed author data
 	 */
-	public readonly author?: APIEmbedAuthor;
+	public get author(): EmbedAuthorData | undefined {
+		if (!this.data.author) return undefined;
+		return {
+			name: this.data.author.name,
+			url: this.data.author.url,
+			iconURL: this.data.author.icon_url,
+			proxyIconURL: this.data.author.proxy_icon_url,
+		};
+	}
 
 	/**
 	 * Received data about the embed provider
 	 */
-	public readonly provider?: APIEmbedProvider;
+	public get provider() {
+		return this.data.provider;
+	}
 
 	/**
 	 * The embed footer data
 	 */
-	public readonly footer?: APIEmbedFooter;
-
-	public constructor(data: APIEmbed = {}) {
-		this.title = data.title;
-		this.description = data.description;
-		this.url = data.url;
-		this.color = data.color;
-		this.thumbnail = data.thumbnail;
-		this.image = data.image;
-		this.video = data.video;
-		this.author = data.author;
-		this.provider = data.provider;
-		this.footer = data.footer;
-		this.fields = data.fields ?? [];
-
-		if (data.timestamp) this.timestamp = new Date(data.timestamp).toISOString();
+	public get footer(): EmbedFooterData | undefined {
+		if (!this.data.footer) return undefined;
+		return {
+			text: this.data.footer.text,
+			iconURL: this.data.footer.icon_url,
+			proxyIconURL: this.data.footer.proxy_icon_url,
+		};
 	}
 
 	/**
 	 * The accumulated length for the embed title, description, fields, footer text, and author name
 	 */
 	public get length(): number {
 		return (
-			(this.title?.length ?? 0) +
-			(this.description?.length ?? 0) +
-			this.fields.reduce((prev, curr) => prev + curr.name.length + curr.value.length, 0) +
-			(this.footer?.text.length ?? 0) +
-			(this.author?.name.length ?? 0)
+			(this.data.title?.length ?? 0) +
+			(this.data.description?.length ?? 0) +
+			(this.data.fields ?? []).reduce((prev, curr) => prev + curr.name.length + curr.value.length, 0) +
```

Comment:
One allocation less:
```suggestion
			(this.data.fields?.reduce((prev, curr) => prev + curr.name.length + curr.value.length, 0) ?? 0) +
```